File: m7-lesson6/um-webapp/src/main/java/com/baeldung/um/spring/UmApp.java
package com.baeldung.um.spring;

import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.builder.SpringApplicationBuilder;

import com.baeldung.um.persistence.setup.MyApplicationContextInitializer;

@SpringBootApplication
public class UmApp {

    public static void main(final String... args) {
        new SpringApplicationBuilder(UmApp.class).initializers(new MyApplicationContextInitializer()).listeners().run(args);
    }

}

File: m12-lesson2/um-webapp/src/main/kotlin/com/baeldung/um/spring/UmApp.kt
package com.baeldung.um.spring

import com.baeldung.um.persistence.setup.MyApplicationContextInitializer
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.builder.SpringApplicationBuilder
import org.springframework.boot.web.servlet.support.SpringBootServletInitializer
import org.springframework.context.annotation.ComponentScan

@SpringBootApplication
@ComponentScan("com.baeldung.um")
class UmApp : SpringBootServletInitializer() {

    override fun configure(application: SpringApplicationBuilder): SpringApplicationBuilder =
            application.sources(UmApp::class.java).initializers(MyApplicationContextInitializer())

}

fun main(args: Array<String>) {
    SpringApplicationBuilder(UmApp::class.java)
            .initializers(MyApplicationContextInitializer())
            .listeners()
            .run(*args)
}